web/src/components/profile/ProfileNewJerseyAddress.test.tsx

```diff
@@ -156,13 +278,78 @@ describe("<ProfileNewJerseyAddress  />", () => {
     const address = generateAddress({
       addressMunicipality: undefined,
     });
+    useMockBusiness(
+      generateBusiness({
+        formationData: generateFormationData({
+          completedFilingPayment: false,
+          formationFormData: generateFormationFormData({ ...address }),
+        }),
+      })
+    );
     renderComponent({
       address,
       municipalities: [municipality],
     });
     blurInput("Address municipality");
     expect(screen.getByText(Config.formation.fields.addressMunicipality.error)).toBeInTheDocument();
   });
+
+  it("renders profile locked fields when completed filing payment is true", async () => {
+    const municipality = generateMunicipality({ displayName: "Allendale", name: "Allendale" });
+    const address = generateAddress({
+      addressLine1: "1111 Home Alone",
+      addressLine2: "Suite 10",
+      addressMunicipality: municipality,
+      addressState: { shortCode: "NJ", name: "New Jersey" },
+      addressZipCode: "00893",
+    });
+    useMockBusiness(
+      generateBusiness({
+        profileData: generateProfileData({ businessPersona: "STARTING" }),
+        formationData: generateFormationData({
+          completedFilingPayment: true,
+          formationFormData: generateFormationFormData({ ...address }),
+        }),
+      })
+    );
+    renderComponent({
+      address,
+      municipalities: [municipality],
+    });
+    const addressLine1 = screen.getByTestId("locked-profileAddressLine1");
+    expect(within(addressLine1).getByText("1111 Home Alone")).toBeInTheDocument();
+    const addressLine2 = screen.getByTestId("locked-profileAddressLine2");
+    expect(within(addressLine2).getByText("Suite 10")).toBeInTheDocument();
+    const addressMuniStateZip = screen.getByTestId("locked-profileAddressMuniStateZip");
+    expect(within(addressMuniStateZip).getByText("Allendale, NJ 00893")).toBeInTheDocument();
+  });
+
+  it("does not render profile locked fields when address fields are not entered and completed filing payment is true", async () => {
```

Comment:
pebble - the way this logic is written, it's basically saying that the input fields will not render and the address will not render, but the header In the Lockfield component will render

Reply:
We talked to Product. This is what they want (see previous comments).